api/v1alpha1/virtualmachine_types.go

```diff
@@ -484,6 +511,15 @@ func (vm VirtualMachine) NamespacedName() string {
 	return vm.Namespace + "/" + vm.Name
 }
 
+// VirtualMachineTemplate defines the specification for configuring VirtualMachine Template
+type VirtualMachineTemplate struct {
+	// NetworkStatus defines the specification of Network
+	Net NetworkStatus `json:"network"`
```

Comment:
It feels a little unnatural to have a member with the `Status` suffix that is used for the specification.  Consider renaming these.

Reply:
yeah makes sense. Changing to `NetworkSpec`